Objective-C/Internal/Replicator/CBLWebSocket.mm

```diff
@@ -603,6 +613,15 @@ - (void) closeWithCode: (C4WebSocketCloseCode)code reason: (NSString*)reason {
 
 // Closes the connection and passes the NSError (if any) to LiteCore.
 - (void) closeWithError: (NSError*)error {
+    CBL_LOCK(self) {
```

Comment:
Sorry I just noticed this. I think there is no need to lock because closeWithError: is always called inside a queue.

Reply:
I can see many flows, some of which are not through queue. Not sure whether they go further back starting from queue.

Reply:
yea, I got this. will update now